infrastructure/modules/aks/providers.tf

```
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "< 4.39.0"
+      version = "< 4.43.0"
```

Still a bug in >= 4.39.0 https://github.com/hashicorp/terraform-provider-azurerm/issues/29772 that hits us.
Keeping the version locked for now
```suggestion
      version = "< 4.39.0"
```